common/flatpak-run.c

```diff
@@ -3198,6 +3198,8 @@ regenerate_ld_cache (GPtrArray    *base_argv_array,
 
 static gboolean
 check_parental_controls (const char    *app_ref,
+                         const char    *app_id,
```

Comment:
I know you’ve probably done it to avoid a `flatpak_decompose_ref()` call, but it seems a bit redundant to pass in both the `app_ref` and the `app_id`, given that the latter is one of the components of the former.

Reply:
Yes, I was on the fence here, I will update it, but maybe I dont need the `app_ref` anymore if I using `g_desktop_app_info_new_from_filename()` + `mct_app_filter_is_appinfo_allowed()`, will check.

Reply:
Ended up keeping `app_ref` and using `flatpak_decompose_ref()` because to filter using the app info we need `app_deploy` (to build the desktop file path) which may be `NULL` here.

Although if the `app_deploy` is `NULL` it means that `app_ref` is for a `runtime`, which I believe we don't handle in malcontent/parental controls (although there is nothing prohibiting one to blacklist a runtime there with `mct_app_filter_builder_blacklist_flatpak_ref()`).

Please let me know what you think and if needed I can remove the separate check for whether the `app_ref` is allowed.

Reply:
LGTM.